### PR TITLE
fix(alerting): Telegram overrides not working as expected

### LINKED ISSUE #919: alerting.telegram.overrides seems to be broken
### Describe the bug

When i add alerting.telegram.overrides, the provider configuration becomes invalid

```
    overrides:
      - group: "something" // redacted
        id: "1100" // redacted
```

Remove the entire override, and it becomes functional again

### What do you see?

2024/11/30 16:33:36 [config.validateAlertingConfig] Ignoring provider=telegram because configuration is invalid

### What do you expect to see?

working provider

### List the steps that must be taken to reproduce this issue

1. add overrides with either token or id or both to telegram provider, it breaks.

### Version

latest tag on docker.

### Additional information

It would be also helpful if the logs said why it is invalid, instead of a blind mention of this config is invalid, thats neither helpful, and wastes a ton of time with trial and error.

### PR DESCRIPTION
**Pull request by baalajimaestro**
Overrides had unexported variables leading to forever invalid configuration

## Summary
Fixes #919 

## Checklist

- [x] Tested and/or added tests to validate that the changes work as intended, if applicable.
- [ ] Updated documentation in `README.md`, if applicable.

### CODE REVIEW

File: alerting/provider/telegram/telegram.go [hunk 1]
```diff
@@ -32,29 +32,29 @@ type AlertProvider struct {
...
+        if len(override.Group) == 0 {
+            return false
+        }
+        if _, ok := registerGroups[override.Group]; ok {
+            return false
+        }
+        registerGroups[override.Group] = true
+    }
+
+    return len(provider.Token) > 0 && len(provider.ID) > 0
```
Comment (TwiN) [inline alerting/provider/telegram/telegram.go:57]: Can you undo this change?

**Review by TwiN — CHANGES_REQUESTED** [review]

File: alerting/provider/telegram/telegram.go (hunk 1 above)
Comment (baalajimaestro) [reply alerting/provider/telegram/telegram.go:57]: The spaces?

Comment (TwiN) [reply alerting/provider/telegram/telegram.go:57]: Yes, indentation in Go is with tabs, not spaces. It was previously indented with tabs, but your IDE must've automatically changed it.

**Comment by TwiN** [comment]
FYI #929 should fix this issue
